airflow/api_connexion/openapi/v1.yaml

```diff
@@ -4659,7 +4642,9 @@ components:
         *Changed in version 2.4.0*&#58; 'sensing' state has been removed.
         *Changed in version 2.4.2*&#58; 'restarting' is added as a possible value
       type: string
+      nullable: true
```

Comment:
This seems to have a side effect. The `state` array in `ListTaskInstanceForm` can now contain null values, and I’m not sure that’s a desired effect (we have `"none"` for that).

Reply:
Indeed, we allow "none" for convenience but I do not find this really consistent not allow 'null' where the TaskInstanceState does. (And a few other endpoint do as well).

It is not easy to work around this in the OpenAPI spec only. (anyOf, allOf, NullRef only etc...).  Those solutions didn't work when I tried them, taking into account the constraint that it needs to be compatible both for `connexion` and `open-api-generator`. We can always duplicate the schema, have one nullable and the original not nullable, but this is not maintainable.

I see 2 solutions moving forward:
1. I have updated the code to allow `null` as a valid value for `ListTaskInstanceForm`. This makes sense to me as we have consistency across endpoints and allowing null alongside 'none' doesn't hurt IMOH.
2. Remove the originally ignored 'nullable: true' in the spec, and handle this later without modifying the actual `State` schema. This would keep a bug in the spec, as most of the ref using `TaskState` expect a nullable field, but the prop is simply ignored. (all but the one in the `ListTaskInstanceForm`)

Also this is an issue for the python client generation. (Client is buggy and also requires some manual patching at release time).

I think this is a small adjustment that will make our life easier, even if allowing both `'none'` and `null` in the form might be a bit overkill.

Let me know if you see a better way to handle this.

Example of query (tested locally):
![image](https://github.com/apache/airflow/assets/14861206/99cb1507-93eb-4cdf-b2a7-01fc0e03336b)

Reply:
Hmm. That makes sense, thanks. This makes me think though, whether the `none` value is needed. It is probably unnecessarily disruptive to remove it outright, but a *Deprecated since* notice in the description can be a good thing?

Reply:
@uranusjr I just added the deprecation warning, thanks for the suggestion.

Reply:
Merging in hope to get it into 2.7.0 and fix 2.7.0 client release.